asf-site-src/source/documentation/in-progress/precommit/index.html.md

```diff
@@ -0,0 +1,164 @@
+<!---
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Basic Precommit
+
+<!-- MarkdownTOC levels="1,2,3" autolink="true" indent="  " bullets="*" bracket="round" -->
+
+* [Purpose](#purpose)
+* [Goals](#goals)
+* [Pre-requisites](#pre-requisites)
+  * [Base Requirements](#base-requirements)
+  * [Plug-ins](#plug-ins)
+    * [Bundled Plug-ins](#bundled-plug-ins)
+    * [Optional Plug-ins](#optional-plug-ins)
+* [More information](#more-information)
+
+<!-- /MarkdownTOC -->
+
+# Purpose
+
+Performing reviews can be an overwhelming process.  The more complex the base, the more comprehensive reviews end up.  Building that functionality into the build itself is a full time job.  Plus, if a new check is added, there is a good chance that the existing code has problems and just want to prevent new bits from making the existing problem worse.
```

Comment:
> Plus, if a new check is added, there is a good chance that the existing code has problems and just want to prevent new bits from

When a new check is added there is a good chance the existing code has problems and often maintainers just want to prevent new bits from

Reply:
Definitely better.